### ISSUE TITLE
no picture after switching channels

### ISSUE THREAD

**Issue by olloESP** [reufer/rpihddevice#1]
... see here on the description and possible fix:
https://www.vdr-portal.de/forum/index.php?thread/133852-rpi-3b-kein-bild-nach-dem-umschalten/&postID=1332648#post1332648

**Comment by dr-m** [comment]
I combined ideas from different contributors in that discussion into this patch.
```diff
diff --git a/omxdevice.c b/omxdevice.c
index c646f90..cf63ad9 100644
--- a/omxdevice.c
+++ b/omxdevice.c
@@ -179,11 +179,14 @@ bool cOmxDevice::SetPlayMode(ePlayMode PlayMode)
 	switch (PlayMode)
 	{
 	case pmNone:
+		if (m_playMode == pmNone) break;
 		FlushStreams(true);
 		m_omx->StopVideo();
 		m_hasAudio = false;
 		m_hasVideo = false;
 		m_videoCodec = cVideoCodec::eInvalid;
+		m_audio->DeInit();
+		m_omx->DeInit();
 		m_playMode = pmNone;
 		break;
 
@@ -191,6 +194,12 @@ bool cOmxDevice::SetPlayMode(ePlayMode PlayMode)
 	case pmAudioOnly:
 	case pmAudioOnlyBlack:
 	case pmVideoOnly:
+		if (m_playMode == pmNone)
+		{
+			m_omx->Init(m_display, m_layer);
+			m_audio->Init();
+			SetVolumeDevice(IsMute() ? 0 : CurrentVolume());
+		}
 		m_playbackSpeed = eNormal;
 		m_direction = eForward;
 		break;
```
I tested this with a patched 1.0.4 (because of #2) on Raspberry OS Legacy, but I do not remember if I tested it without the patch. I tested switching between DVB-T and DVB-T2 (H.264) channels, same or different transponder.

Since then, I switched to Debian Bookworm but did not get it running yet (see #3).

**Comment by dr-m** [comment]
I switched back from Debian Bookworm to Raspberry OS Legacy and tested VDR 2.6.0 and rpihddevice 1.0.5 without the above patch.

Without the patch, I will indeed get no picture after switching channels, only audio and a black screen with OSD. With the patch, it works fine.
